DXRModules/DeusEx/Classes/DXRFixup.uc

```diff
@@ -797,6 +806,38 @@ static function FixConversationAddNote(Conversation c, string textSnippet)
     }
 }
 
-defaultproperties
+function SetLampState(#var(prefix)Lamp lmp, bool turnOn) // could maybe go in DXRLamp if it existed
+{
+    if (turnOn) {
+        lmp.bOn = True;
+        lmp.LightType = LT_Steady;
+        // lmp.PlaySound(sound'Switch4ClickOn');
+        lmp.bUnlit = True;
+        lmp.ScaleGlow = 3.0;
+    } else if (lmp.bOn) {
+        lmp.bOn = False;
+        lmp.LightType = LT_None;
+        // lmp.PlaySound(sound'Switch4ClickOff');
+        lmp.bUnlit = False;
+        lmp.ResetScaleGlow();
+    }
+}
+
+// the Plane struct is repurposed here to describe a disk orthogonal to the xy-plane
+function SetAllLampsState(bool type1, bool type2, bool type3, optional Plane p, optional bool applyToModifiedMaps)
 {
+    local #var(prefix)Lamp lmp;
+
+    if (class'MenuChoice_AutoLamps'.default.enabled == false || (!applyToModifiedMaps && #defined(revision || gmdx)))
+        return;
+
+    if (p.W == 0.0) {
+        foreach AllActors(class'#var(prefix)Lamp', lmp) {
+            SetLampState(lmp, (Lamp1(lmp) != None && type1) || (Lamp2(lmp) != None && type2) || (Lamp3(lmp) != None && type3));
+        }
+    } else {
+        foreach RadiusActors(class'#var(prefix)Lamp', lmp, p.W, p) {
```

Comment:
The location you are using isn't being vectm'd, so this will be centred on the wrong location on a mirrored map, won't it?

Reply:
Plane derives from Vector, so my guess is there wouldn't be a problem there (or it wouldn't be difficult so solve).